radis/lbl/base.py

```diff
@@ -1526,6 +1539,52 @@ def _calc_noneq_parameters(self, vib_distribution, singleTvibmode):
             self.calc_weighted_trans_moment()
             self.calc_einstein_coefficients()
 
+        # Caching the non-equilibrium parameters Evib and Erot
+        # First, getting the Electronic States file through partition functions:
+        for isotope in self._get_isotope_list(molecule):
+            elec_state = self.get_partition_function_calculator(
+                molecule, isotope, state
+            ).ElecState
+            # fname = "{0}_{1}_{2}_EvibErot".format(elec_state.jsonfile, molecule, isotope)
+            fname = elec_state.jsonfile
+            f1 = splitext(elec_state.jsonfile)[0] + "_{0}_{1}_EvibErot".format(
```

Comment:
This is the right time to save the energies, but you partially misunderstood the structure of the code. 

There are 2 database files : line database (loaded in df) and energy levels, either given directly or computed from spectroscopic constants. In noneq calculations we parse through the line database and, for each line, which is a transition between a lower and upper level, we look up the Energy database and assign the corresponding energy of the lower level, and the energy of the  upper level. This look-up takes time, too much time, your job is to cache the results (unless user uses a different Energy database) 

Therefore, the cache file should be created next to the Line database. Here you generated it next to the spectroscopic constants.

Have a look at the Radis Architecture graph in the developer section of the documentation. It will help.

Also, I think you should only be saving the newly added columns. What about  : 
- creating a database file ORIGINAL_FILE_extra_columns.hdf5
-  after initial look up of database, line 1513,if it fails `if not all_in(required_columns, df)` then add a new step where you'll look up if there is an _extra_column file. If so and if it has the right columns, and the right lines (same index).
-- when you save and create the cache file make sure you save the index. Imagine you're computing the default CO spectrum on a limited range. If computing afterwards on a larger rznge it should fail and regenerate the cache. If computing again on a limited range included in the larger range, it should work.
- you'll need the paths to the initial database. Look up the fetch_databank code. The dbpath parameter may help.

Reply:
Thanks a lot for the update! So right now the cache files being generated correspond to the energy database which takes up a lot of time in calculations, got it.
- If I'm not wrong, the line databases are stored in `radis.config["DEFAULT_DOWNLOAD_PATH"]/molecule`. So, I need to save the cached files in this folder(probably deeper in the subdirectories, I have yet to fully figure that out).

the `ORIGINAL_FILE_extra_columns.hdf5` needs to be generated in the directory mentioned above, with the relevant columns, which should be `df["Evibl"], df["Evibu"], df["Erotl"] and df["Erotu"]`.

Reply:
Line databases are stored in HOME/.radisdb

Reply:
You might have access to the database files in `self.params.dbpath`  (not sure)